source/core/slang-io.cpp

```diff
@@ -137,13 +137,40 @@ namespace Slang
 	String Path::combine(const String& path1, const String& path2)
 	{
 		if (path1.getLength() == 0) return path2;
-		StringBuilder sb(path1.getLength()+path2.getLength()+2);
-		sb.Append(path1);
-		if (!path1.endsWith('\\') && !path1.endsWith('/'))
-			sb.Append(kPathDelimiter);
-		sb.Append(path2);
-		return sb.ProduceString();
+
+        StringBuilder sb;
+        combineBuilder(path1.getUnownedSlice(), path2.getUnownedSlice(), sb);
+        return sb.ProduceString();
 	}
+    /* static */void Path::combineBuilder(const UnownedStringSlice& path1, const UnownedStringSlice& path2, StringBuilder& outBuilder)
```

Comment:
I'm not 100% happy with this name, since this isn't a function that combines a builder (as its name implies), but rather a version of `Path::combine` that combines *into* a builder. So maybe `combineIntoBuilder`? Or even just `combineInto`?

Realistically, the `outBuilder` argument could just be `outString` since `StringBuilder` is really just a convenience wrapper on `String`. And at that point should be just have  it be `Path::combineInPlace(String& ioPath1, const String& path2)`?

Reply:
Fair enough on combineIntoBuilder. 

I'm not sure I'd describe a StringBuilder being very 'wrapper-like'. I either want to use the builders extra facilities (and other assumptions), or a I want the container of a result - the String. I can't directly create a StringBuilder from a String or swap them (and so keep the memory usage). I can't assign a string to a StringBuilder. Perhaps I'm missing something, but they don't seem very 'interchangeable' - which is perhaps fine or even good, but doesn't seem to be what the comment implies. 

Anyway with that in mind the 'combine in place style' would seem to want the first parameter to be StringBuilder, and that is desirable.